src/axes/categoryAxis.ts

```diff
@@ -242,40 +256,47 @@ namespace Plottable.Axes {
      * @param {Plottable.Scales.Category} scale The scale this axis is representing.
      * @param {string[]} ticks The strings that will be printed on the ticks.
      */
-    private _measureTickLabels(axisWidth: number, axisHeight: number, scale: Scales.Category, ticks: string[]) {
-      let axisSpace = this._isHorizontal() ? axisWidth : axisHeight;
-      let totalOuterPaddingRatio = 2 * scale.outerPadding();
-      let totalInnerPaddingRatio = (ticks.length - 1) * scale.innerPadding();
-      let expectedRangeBand = axisSpace / (totalOuterPaddingRatio + totalInnerPaddingRatio + ticks.length);
-      let stepWidth = expectedRangeBand * (1 + scale.innerPadding());
+    private _measureTickLabels(axisWidth: number, axisHeight: number) {
+      const thisScale = <Scales.Category> this._scale;
 
-      let wrappingResults = ticks.map((s: string) => {
+      // set up a test scale to simulate rendering ticks with the given width and height.
+      const scale = new Scales.Category()
+        .domain(thisScale.domain())
+        .innerPadding(thisScale.innerPadding())
+        .outerPadding(thisScale.outerPadding())
+        .range([0, this._isHorizontal() ? axisWidth : axisHeight]);
 
-        // HACKHACK: https://github.com/palantir/svg-typewriter/issues/25
-        let width = axisWidth - this._tickSpaceRequired(); // default for left/right
-        if (this._isHorizontal()) { // case for top/bottom
-          width = stepWidth; // defaults to the band width
-          if (this._tickLabelAngle !== 0) { // rotated label
-            width = axisHeight - this._tickSpaceRequired(); // use the axis height
-          }
-          // HACKHACK: Wrapper fails under negative circumstances
-          width = Math.max(width, 0);
-        }
+      const [ticks, stepWidth] = this.getDownsampleInfo(scale);
 
-        // HACKHACK: https://github.com/palantir/svg-typewriter/issues/25
-        let height = stepWidth; // default for left/right
-        if (this._isHorizontal()) { // case for top/bottom
-          height = axisHeight - this._tickSpaceRequired();
-          if (this._tickLabelAngle !== 0) { // rotated label
-            height = axisWidth - this._tickSpaceRequired();
-          }
-          // HACKHACK: Wrapper fails under negative circumstances
-          height = Math.max(height, 0);
+      // HACKHACK: https://github.com/palantir/svg-typewriter/issues/25
+      // the width (x-axis specific) available to a single tick label.
+      let width = axisWidth - this._tickSpaceRequired(); // default for left/right
+      if (this._isHorizontal()) { // case for top/bottom
+        width = stepWidth; // defaults to the band width
+        if (this._tickLabelAngle !== 0) { // rotated label
```

Comment:
Imo that's just more verbose but with no gain. Regardless I want to rework large parts of this since it's so hacky so might make more sense to tackle things then